### ISSUE TITLE
WebExtensions platform

### ISSUE THREAD

**Issue by ruv** [palant/enforceencryption#13]
Do you plan to move on the WebExtensions API?

For the moment, Enforce Encryption 1.0.7 is marked as "legacy" in Firefox 56 and it is disabled in Firefox 57.

> [Note](https://support.mozilla.org/en-US/kb/firefox-add-technology-modernizing): In Firefox version 57, in release on November 14, 2017, only extensions built with this new technology will work in Firefox.

**Comment by palant** [comment]
Porting was blocked on https://bugzilla.mozilla.org/show_bug.cgi?id=1368527. With that bug resolved in Firefox 57, my port might actually be working in Firefox now - I need to test it again and see whether there are more issues.

**Comment by palant** [comment]
No luck. Determining current state works correctly now but actually enforcing encryption doesn't. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1418275 on that.

**Comment by palant** [comment]
I pushed my current state, currently it only works in Chrome however. The Firefox bug will hopefully be fixed soon.

**Comment by palant** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1418275 is fixed but it doesn't quite do for us - we set headers on a cached request which doesn't work. I created https://bugzilla.mozilla.org/show_bug.cgi?id=1431043 on the new issue.

In the meantime, I guess that we can continue here. Hitting the network whenever users want to enforce encryption is suboptimal but it is better than nothing. We can have this code dependent on the browser and still produce a cached request in Chrome.
